File: Data Structures & Algorithms/is-palindrome/submission-2.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        length = len(s)
        cur = 0
        end = length - 1

        while end > cur:
            while end > cur and not s[end].isalnum():
                end -= 1
            while end > cur and not s[cur].isalnum():
                cur += 1
            if s[cur].lower() != s[end].lower():
                return False
            end -= 1
            cur += 1
        return True

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> s;
        for(string t:tokens){
            if(isalnum(t[t.length()-1])){
                s.push(stoi(t));
            } else {
                int second = s.top();
                s.pop();
                int first = s.top();
                s.pop();

                if(t == "+"){
                    s.push(first + second);
                } else if(t == "-"){
                    s.push(first - second);
                } else if(t == "*"){
                    s.push(first * second);
                } else { // '/' (divide)
                    s.push(first / second);
                }
            }
        }

        return s.top();
    }
};
